txscript/data/script_tests.json

```diff
@@ -1238,9 +1238,104 @@
+["", "ABS 0 EQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "ABS requires an item on the stack, stack has no items"],
+["0 ABS", "0 EQUAL", "NONE", "OK"],
+["16 ABS", "16 EQUAL", "NONE", "OK"],
+["-16 ABS", "16 EQUAL", "NONE", "OK", "ABS must replace the numeric stack item with its absolute equivalent"],
+["2147483648 ABS", "2147483648 EQUAL", "NONE", "ERR_OUT_OF_RANGE", "ABS must fail with stack item >4 bytes"],
+
+["NOT test coverage"],
+["", "NOT 0 EQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "NOT requires an item on the stack, stack has no items"],
+["NOP", "NOT 1 EQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "NOT requires an item on the stack and NOP must not be treated as one"],
+["0 NOT", "NOP", "NONE", "OK"],
+["1 NOT", "0 EQUAL", "NONE", "OK", "NOT must invert the boolean value of a numeric stack item (positive numeric case)"],
+["11 NOT", "0 EQUAL", "NONE", "OK"],
+["-11 NOT", "0 EQUAL", "NONE", "OK", "NOT must invert the boolean value of a numeric stack item (negative numeric case)"],
+["'abcdef' NOT", "0 EQUAL", "NONE", "ERR_OUT_OF_RANGE", "NOT is an arithmetic operand"],
+
+["0NOTEQUAL test coverage"],
+["", "0NOTEQUAL 0 EQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "0NOTEQUAL requires an item on the stack, stack has no items"],
+["NOP", "0NOTEQUAL 1", "NONE", "ERR_INVALID_STACK_OPERATION", "0NOTEQUAL requires an item on the stack and NOP must not be treated as one"],
+["0 0NOTEQUAL", "0 EQUAL", "NONE", "OK"],
+["1 0NOTEQUAL", "1 EQUAL", "NONE", "OK", "0NOTEQUAL must assert if a numeric stack item is not equal to zero."],
+["111 0NOTEQUAL", "1 EQUAL", "NONE", "OK"],
+["-111 0NOTEQUAL", "1 EQUAL", "NONE", "OK"],
+["2147483648", "0NOTEQUAL 0 EQUAL", "NONE", "ERR_OUT_OF_RANGE", "0NOTEQUAL must fail with stack item >4 bytes"],
+
+["ADD test coverage"],
+["NOP", "ADD 0 EQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "ADD requires two items on the stack and NOP must not be treated as one"],
+["", "ADD 0 EQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "ADD requires two items on the stack, stack has no items"],
+["0", "ADD 0 EQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "ADD requires two items on the stack, stack has one item"],
+["2 -2 ADD", "0 EQUAL", "NONE", "OK", "ADD must add the first and second stack items"],
+["2147483647 -2147483647 ADD", "0 EQUAL", "NONE", "OK"],
+["2147483647 DUP ADD", "4294967294 EQUAL", "NONE", "OK", ">32 bit EQUAL is valid"],
+["2147483647 NEGATE DUP ADD", "-4294967294 EQUAL", "NONE", "OK"],
+["-1 -1 ADD", "-2 EQUAL", "NONE", "OK"],
+["1", "ADD", "NONE", "ERR_INVALID_STACK_OPERATION"],
+["2147483648 0 ADD", "NOP", "NONE", "ERR_OUT_OF_RANGE", "Arithmetic operands must be in range [-2^31...2^31] (positive numeric case)"],
+["-2147483648 0 ADD", "NOP", "NONE", "ERR_OUT_OF_RANGE", "Arithmetic operands must be in range [-2^31...2^31] (negative numerica case)"],
+["2147483648 1", "ADD 2147483649 EQUAL", "NONE", "ERR_OUT_OF_RANGE", "ADD must fail with first operand >4 bytes"],
```

Comment:
Comment is incorrect.  The first operand to the opcode is the second one in the script because they are popped in reverse order.

Same is true with all of these.